sqlite3_opt_math_functions_test.go

```diff
@@ -0,0 +1,28 @@
+// +build sqlite_math_functions
+
+package sqlite3
+
+import (
+	"database/sql"
+	"testing"
+)
+
+func TestMathFunctions(t *testing.T) {
+	db, err := sql.Open("sqlite3", ":memory:")
+	if err != nil {
+		t.Fatal("Failed to open database:", err)
+	}
+	defer db.Close()
+
+	queries := []string{
+		`SELECT acos(1)`,
+		`SELECT log(10, 100)`,
+		`SELECT power(2, 2)`,
+	}
+
+	for _, query := range queries {
+		if _, err := db.Query(query); err != nil {
```

Comment:
Right now you are dropping the `Rows` object. That means you are only checking for an error from `sqlite3_prepare_v2` and not `sqlite3_step`. For completeness we should check both, which is why `QueryRow` would be preferable.

```go
var result float64
if err := db.QueryRow(query).Scan(&result); err != nil {
    t.Errorf("invoking math function query %q: %v", query, err)
}
// For the purposes of this test, we don't care about the output, just that the query succeeded.
```

Reply:
I see... thanks for the review :)